### ISSUE TITLE
Impossible to add carddav addressbook in RoundCube (baikal)

### ISSUE THREAD

**Issue by pousterlus** [blind-coder/rcmcarddav#142]
Hello,

I searched all the forums and issues lists everywhere... but I can't find the solution.
I've installed the plugin in RoundCube on my FreeBSD 10 with Baikal, I correctly see the CardDAV options in the Settings tab.
But I can't add anything, it seems like when I click on "Save", every fields are blanked.

So I've tried to add the informations in the config.inc.php inside the carddav directory :
```
// Each addressbook preset takes the following form:
/*
$prefs['baikal'] = array(
        // required attributes
        'name'         =>  'CardDAV',
        'username'     =>  'gorb',
        'password'     =>  'xxxxx',
        'url'          =>  'https://dav.mydomain.net:443/card.php/addressbooks/gorb/default/',

        // optional attributes
        'active'       =>  true,
        'readonly'     =>  false,
        'refresh_time' => '00:05:00',

        // attributes that are fixed (i.e., not editable by the user) and
        // auto-updated for this preset
        'fixed'        =>  array( 'username', 'password' ),

        // hide this preset from CalDAV preferences section so users can't even
        // see it
        'hide' => false,
);
*/
```
But nothing happens in the Address Book tab list : I only have the default roundcube addressbook.

If anyone can point me out where to look for a mistake I could have made, it would be greatly appreciated ! :)

Thanks,

-- 
Léo, Paris - France.

**Comment by pousterlus** [comment]
Hello again,
Maybe that's because of the mysql init script.
I just figured out that there's a problem : 

```
[root@concorde /usr/local/www/roundcube/plugins/carddav/dbmigrations/0000-dbinit]# mysql -u root -p < mysql.sql roundcube
Enter password: xxxx
ERROR 1215 (HY000) at line 2: Cannot add foreign key constraint
```
Maybe the dbinit script doesn't work with "mysql  Ver 14.14 Distrib 5.7.10, for FreeBSD10.2 (amd64) using  EditLine wrapper" ?

-- 
Léo, Paris - France.

**Comment by blind-coder** [comment]
The 0000-dbinit script is executed automatically upon first login, so there should be no problem. I also highly doubt that there is a version incompatibility there.
On your first post, you do know that the preset is commented out (using /* */ syntax), so it won't be executed.
Please check the log files in roundcube/logs/carddav*

**Comment by pousterlus** [comment]
Hello,
I must have been really tired when I tried to set your plugin up...
That was, not surprisingly, the "/*" comment at the beginning of the address book's definition in the config.inc.php :)

I uncommented it, and.. tada ! The CardDav has shown up in Address Book tab.
But, when I try to create a user with the "+" icon, I hit the "Save" button and nothing happens.

In the logs, i found this : 
`[15-Feb-2016 16:45:20 +0100]: <3p8l76d6> BACKEND: (carddavconfig) FATAL! Request for non-existent configuration 3`
Seems like a DB issue ?

Thanks,

-- 
Léo, Paris - France.
